modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokerImpl.java

```diff
@@ -153,7 +154,7 @@ protected DDMDataProviderResponse doInvokeExternal(
 					DDMRESTDataProviderSettings.class));
 
 		DDMDataProviderResponse ddmDataProviderResponse =
-			ddmDataProviderInvokeCommand.execute();
+			ddmDataProviderInvokeCommand.run();
```

Comment:
@vngoliveira the code looks good, but could you explain to me why using the `run` method fixes the issue?

Reply:
Hey @carolmariaabb
From what I understand, to work correctly in a new instance, we need to set up the Hystrix thread to connect to the new instance. When you call the run method, it filters the instance in the getData method and gets the data.

Reply:
I'm a little worried about this change. I was reading the main differences between these methods and here is what I found:

- Use execute() when you want to perform the command's action asynchronously without blocking the current thread. This is useful for long-running or I/O-bound operations.

- Use run() when you need the result of the command immediately and are willing to block the current thread. This is suitable for short-running operations where blocking is acceptable.

Therefore, with your changes, we would change the current behavior of the Data Provider.

- What error is shown when trying to reproduce the issue without the fix?
- Did you try to reproduce the same issue without the `include-and-override=portal-liferay-online.properties in portal-ext.properties` property?